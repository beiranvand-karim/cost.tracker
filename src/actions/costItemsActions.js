export const ADD_FILTERED_COST_ITEMS = "ADD_FILTERED_COST_ITEMS";
export const ADD_COST_ITEM = "ADD_COST_ITEM";


export const addFilteredCostItemsAction = data => ({
  type: ADD_FILTERED_COST_ITEMS,
  payload: data
});

export const addCostItemAction = data => ({
  type: ADD_COST_ITEM,
  payload: data
});
