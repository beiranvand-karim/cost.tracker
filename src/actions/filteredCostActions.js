export const ADD_COST = 'ADD_COST';
export const ADD_FILTERED_COSTS = 'ADD_FILTERED_COSTS';

export const addCostAction = (data) => {
   return {
      type: ADD_COST,
      payload: data
   }
};

export const addFilteredActions = (data) => {
   return {
      type: ADD_FILTERED_COSTS,
      payload: data
   }
};
