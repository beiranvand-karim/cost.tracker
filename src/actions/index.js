export {
  addCostItemAction,
  addFilteredCostItemsAction
} from "./costItemsActions";
export { addSourceCostItemAction } from "./sourceCostItemsActions";
