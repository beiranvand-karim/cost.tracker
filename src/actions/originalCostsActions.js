export const ADD_ORIGINAL_COST = 'ADD_ORIGINAL_COST';

export const addOriginalCostAction = (data) => {
   return {
      type: ADD_ORIGINAL_COST,
      payload: data
   }
};
