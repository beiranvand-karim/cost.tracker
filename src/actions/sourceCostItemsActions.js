export const ADD_SOURCE_COST_ITEM = "ADD_SOURCE_COST_ITEM";

export const addSourceCostItemAction = data => ({
  type: ADD_SOURCE_COST_ITEM,
  payload: data
});
