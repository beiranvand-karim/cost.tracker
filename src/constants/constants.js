export const NO_FILTER = "NO_FILTER";
export const DUMMY_ACTION = "DUMMY_ACTION";
