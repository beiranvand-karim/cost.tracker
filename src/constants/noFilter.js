const NO_FILTER = 'NO_FILTER';

export default NO_FILTER;
