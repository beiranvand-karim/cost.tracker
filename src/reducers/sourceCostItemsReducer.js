import { ADD_SOURCE_COST_ITEM } from "../actions/sourceCostItemsActions";

const defaultState = {
  costItems: []
};

export default function sourceCostItemsReducer(state = defaultState, action) {
  switch (action.type) {
    case ADD_SOURCE_COST_ITEM:
      const { costItems } = state;
      costItems.unshift(action.payload);
      return {
        ...state,
        costItems
      };
    default:
      return state;
  }
}
