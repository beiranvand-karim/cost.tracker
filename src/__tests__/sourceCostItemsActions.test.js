/* global expect, describe, it, beforeEach */
import { ADD_SOURCE_COST_ITEM } from "../actions/sourceCostItemsActions";
import sourceCostItemsReducer from "../reducers/sourceCostItemsReducer";
import { DUMMY_ACTION } from "../constants/constants";
import costItems from "../__helpers__/costItems";

describe("sourceCostItemsReducer", () => {
  it("should matches the initial state", () => {
    const action = { type: DUMMY_ACTION };
    expect(sourceCostItemsReducer(undefined, action)).toMatchSnapshot();
  });

  it("should matches the add original cost state", () => {
    const action = {
      type: ADD_SOURCE_COST_ITEM,
      payload: costItems
    };
    expect(sourceCostItemsReducer(undefined, action)).toMatchSnapshot();
  });
});
