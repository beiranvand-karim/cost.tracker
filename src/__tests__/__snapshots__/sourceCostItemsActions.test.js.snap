// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sourceCostItemsReducer should matches the add original cost state 1`] = `
Object {
  "costItems": Array [
    Array [
      Object {
        "category": "CLOTHING",
        "comment": "this is a comment",
        "cost": 1000,
      },
      Object {
        "category": "RENT",
        "comment": "this is a comment",
        "cost": 1500,
      },
      Object {
        "category": "FOOD",
        "comment": "this is a comment",
        "cost": 2000,
      },
    ],
  ],
}
`;

exports[`sourceCostItemsReducer should matches the initial state 1`] = `
Object {
  "costItems": Array [],
}
`;
